react-basic-practice/src/components/Quantity/index.jsx

```diff
@@ -18,11 +19,15 @@ const Quantity = (props) => {
 
   const decrementCount = () => {
     // Update state with decremented value
-    if (counter > 0) {
+    if (counter > 1) {
       setCounter((preCounter) => preCounter - 1);
     }
   };
 
+  const setQuantity = (value) => {
+    setCounter(value.quantity);
+  };
+
   if (!status) {
```

Comment:
Please explain this.

Reply:
`status` is the status of the existing product, if it is `sold out`, the data will not be obtained from the quantity button, if the status is false, the value can be obtained from the quantity button

Reply:
You are mixing data types. If it's string then all values must be string, you should never have many data types for the same prop or variable.